packages/wrangler/src/entry.ts

```diff
@@ -25,20 +25,34 @@ export async function getEntry(
 ): Promise<Entry> {
   let file: string;
   let directory = process.cwd();
+  let isSiteEntryPoint = false;
   if (args.script) {
     // If the script name comes from the command line it is relative to the current working directory.
     file = path.resolve(args.script);
   } else if (config.main === undefined) {
-    throw new Error(
-      `Missing entry-point: The entry-point should be specified via the command line (e.g. \`wrangler ${command} path/to/script\`) or the \`main\` config field.`
-    );
+    if (config.site?.["entry-point"]) {
+      isSiteEntryPoint = true;
+      directory = path.resolve(path.dirname(config.configPath ?? "."));
+      file = path.resolve(config.site?.["entry-point"]);
+    } else {
+      throw new Error(
+        `Missing entry-point: The entry-point should be specified via the command line (e.g. \`wrangler ${command} path/to/script\`) or the \`main\` config field.`
+      );
+    }
   } else {
     directory = path.resolve(path.dirname(config.configPath ?? "."));
     file = path.resolve(directory, config.main);
   }
 
   await runCustomBuild(file, config.build);
 
+  if (isSiteEntryPoint) {
+    // site.entry-point could be a directory
+    if (existsSync(file) && lstatSync(file).isDirectory()) {
+      file = path.resolve(file, "index.js");
+    }
+  }
+
```

Comment:
Should this be above the custom build? I.E. passing this transformed `file` to the custom build?

Reply:
No, the custom build will likely generate the file to be used as an entry

Reply:
oh hmm, but it does expect a full path to the expected file.

Reply:
sigh, ok gotta hack it with the extname thing instead of actually checking whether it's a dir